srcs/juloo.keyboard2/EmojiGridView.java

```diff
@@ -110,6 +111,33 @@ SharedPreferences emojiSharedPreferences()
     return getContext().getSharedPreferences("emoji_last_use", Context.MODE_PRIVATE);
   }
 
+  private void migrateOldPrefs()
```

Comment:
This does some work that is re-done quickly after and loose the result (`emojiSharedPreferences`, `getStringSet`).
What do you think of changing that into a function that returns the last used (eg. `Set<String> loadLastUsedFromPrefs()`) and would be called from `loadLastUsed()` ?
That would make this function the only place where the prefs are loaded, ensuring migration is always checked.